src/hiero_sdk_python/query/query.py

```
@@ -111,6 +142,20 @@ def _before_execute(self, client: Client) -> None:
+            # if max_query_payment not set fall back to the client-level default max query payment
+            max_payment = (
+                self.max_query_payment 
+                if self.max_query_payment is not None 
+                else client.default_max_query_payment
+            )
+            
+            if self.payment_amount > max_payment:
+                raise ValueError(
+                    f"Query cost ℏ{self.payment_amount.to_hbars()} HBAR "
+                    f"exceeds max set query payment: ℏ{max_payment.to_hbars()} HBAR"
+                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)